package/kedro_viz/data_access/managers.py

```diff
@@ -82,6 +82,22 @@ def add_catalog(self, catalog: DataCatalog):
             if self.tracking_datasets.is_tracking_dataset(dataset):
                 self.tracking_datasets.add_tracking_dataset(dataset_name, dataset)
 
+    def resolve_dataset_factory_patterns(
+        self, catalog: DataCatalog, pipelines: Dict[str, KedroPipeline]
+    ):
+        """Resolve dataset factory patterns in data catalog by matching
+        them against the datasets in the pipelines.
+        """
+        for pipeline in pipelines.values():
+            # Temporary try/except block so the Kedro develop branch can work with Viz.
+            try:
```

Comment:
This might catch other exceptions. You could try `if hasattr(pipeline, 'data_set`):`